### PR TITLE
MOTECH-2791: Jasmine tests for Scheduler

### PR DESCRIPTION
**Pull request by palbecki**

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by jredlarski** [comment]
test this please

File: src/scheduler/create-or-update-job.controller.spec.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            expect(scope.getMinDate("RUN_ONCE")).toBeTruthy();
+        });
+
+        it('should return null', function () {
+            expect(scope.getMinDate()).toBeNull();
+        });
+    });
+
+    describe('create or update job method', function () {
+        it('should execute MotechAlert with with good parameters if scope.dates.startDate greater than scope.dates.endDate', function () {
```
Comment (jredlarski) [inline src/scheduler/create-or-update-job.controller.spec.js]: @palbecki remove redundant 'with'

File: src/scheduler/create-or-update-job.controller.spec.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            scope.dates.endDate = scope.parseDateToString(currentDate.addDays(1));
+
+            scope.createOrUpdateJob();
+
+            expect(MotechAlertSpy).not.toHaveBeenCalledWith(
+                i18nService.getMessage("scheduler.error.endDateBeforeStartDate",
+                                       [scope.dates.startDate, scope.dates.endDate]), "scheduler.error");
+        });
+
+        it('should not execute MotechAlert with with good parameters if scope.dates.endDate greater than scope.dates.startDate', function () {
```
Comment (jredlarski) [inline src/scheduler/create-or-update-job.controller.spec.js]: rename the test to: 'should not execute MotechAlert with good parameters if scope.dates.startDate is after scope.dates.endDate'

File: src/scheduler/create-or-update-job.controller.spec.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            spyOn(JobsService, 'createJob');
+
+            scope.createOrUpdateJob();
+            scope.createOrUpdateJob('edit');
+
+            expect(LoadingModal.open).not.toHaveBeenCalled();
+            expect(JobsService.createJob).not.toHaveBeenCalled();
+        });
+
+        it('should show confirm if if action is "edit"', function () {
```
Comment (jredlarski) [inline src/scheduler/create-or-update-job.controller.spec.js]: remove redundant `if`

File: src/scheduler/create-or-update-job.controller.spec.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            spyOn(ModalFactory, 'showConfirm');
+
+            scope.createOrUpdateJob('edit');
+
+            expect(ModalFactory.showConfirm).toHaveBeenCalled();
+        });
+
+        it('should not show confirm if if action is not "edit"', function () {
```
Comment (jredlarski) [inline src/scheduler/create-or-update-job.controller.spec.js]: remove redundant `if`

File: src/scheduler/period-validity.directive.spec.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                '<input type="text" ng-model="model.period" name="period" period-validity />' +
+            '</form>'
+        );
+        scope.model = { period: null };
+        $compile(element)(scope);
+        scope.$digest();
+        form = scope.form;
+    }));
+
+    describe('string', function() {
```
Comment (jredlarski) [inline src/scheduler/period-validity.directive.spec.js]: Shouldn't it be something different than 'string'? I believe that we can delete this describe line and leave the test as they are

**Comment by jredlarski** [comment]
test this please

**Pull request merged**